### PR TITLE
カテゴリフィルタの空状態を追加

### LINKED ISSUE #1204: カテゴリフィルタ適用時の空状態と統合テストを追加する
## 関連Issue

#1200

## 概要

カテゴリフィルタ適用時の空状態と統合テストを追加する。

該当する支払いがない場合に、フィルタ中であることが分かる表示または解除導線を整える。

## ToDo

- [ ] カテゴリ条件に該当する支払いがない場合の空状態を表示する
- [ ] フィルタ中であることが分かる表示または解除導線を追加する
- [ ] Page story でカテゴリあり・カテゴリ未設定・空結果を確認する
- [ ] integration test でユーザー操作によるカテゴリフィルタ適用を確認する
- [ ] 既存の支払い作成・削除・詳細表示のテストが壊れていないことを確認する

## 完了の定義 (Definition of Done)

- カテゴリフィルタの空結果が画面上で分かる
- 空結果からフィルタ解除できる、または解除方法が分かる
- ユーザー操作でカテゴリありの絞り込み結果を確認できる
- ユーザー操作でカテゴリ未設定の絞り込み結果を確認できる
- Web の lint / format-check / typecheck / unit test / storybook test が通る

## 実装方針・技術的メモ

- カテゴリフィルタ UI Issue の後に実装する
- Page story は既存の PaymentsPage / PaymentList 周辺の構成に寄せる
- 追加の検索条件は扱わない

## 補足

画面イメージ:

カテゴリ条件に一致する支払いがない場合:

```text
[食費]     [Create payment]

該当する支払いはありません
[フィルタを解除する]
```

カテゴリ未設定に一致する支払いがない場合:

```text
[未設定]   [Create payment]

該当する支払いはありません
[フィルタを解除する]
```

### PR DESCRIPTION
**Pull request by kosnu**
## 関連Issue

- #1204

## 変更内容

- カテゴリフィルタ適用中に支払いが0件の場合の空状態と解除導線を追加した
- ユーザー操作でカテゴリあり・カテゴリ未設定・空結果解除を確認するテストを追加した
- 支払い0件時はカテゴリ取得完了を待たずに空状態を表示するようにした

## 動作確認

- [x] task web:lint
- [x] task web:format-check
- [x] task web:typecheck
- [x] task web:test-unit
- [x] task web:test-storybook

## 補足

- PaymentsPage の story 追加は含めていません。

### CODE REVIEW

File: apps/web/src/features/payments/listPayment/PaymentList/PaymentList.tsx [hunk 1]
```diff
@@ -93,6 +117,25 @@ const Items = memo(function Body({ promiseCategories, getPayments, onOpenPayment
...
+}: {
+  filtered: boolean
+  onClearCategory: () => void
+}) {
+  return (
+    <Flex align="start" direction="column" gap="2">
+      <Text color="gray">No payments found.</Text>
+      {filtered ? (
+        <Button variant="soft" onClick={onClearCategory}>
+          Clear filter
```
[the remark this answers is not in the document]
Comment (kosnu) [reply apps/web/src/features/payments/listPayment/PaymentList/PaymentList.tsx:132]: まだそのアクセシビリティは考えていないので流します

File: apps/web/src/features/payments/listPayment/PaymentList/PaymentList.tsx [hunk 2]
```diff
@@ -93,6 +117,25 @@ const Items = memo(function Body({ promiseCategories, getPayments, onOpenPayment
...
+        </Button>
+      ) : null}
```
[the remark this answers is not in the document]
Comment (kosnu) [reply apps/web/src/features/payments/listPayment/PaymentList/PaymentList.tsx:134]: Handled in commit  dc91aa6d  by adding a PaymentList test for the unfiltered empty state. It verifies that No payments found. is shown and the Clear filter button is not rendered when category is absent from the search params.
